rudder-lang/src/parser/error.rs

```diff
@@ -166,30 +184,103 @@ where
     E: Fn() -> PErrorKind<PInput<'src>>,
 {
     move |input| {
-        let x = f(input);
-        match x {
+        match f(input) {
             // a non nom error cannot be superseded
             // keep original context when possible
             Err(Err::Failure(err)) => match err.kind {
                 PErrorKind::Nom(_) => Err(Err::Failure(PError {
-                    context: err.context,
+                    context: None,
                     kind: e(),
                 })),
                 _ => Err(Err::Failure(err)),
             },
-            Err(Err::Error(err)) => Err(Err::Failure(PError {
-                context: err.context,
+            Err(Err::Error(_)) => Err(Err::Failure(PError {
+                context: None,
                 kind: e(),
             })),
-            Err(Err::Incomplete(_)) => Err(Err::Failure(PError {
-                context: input,
+            Err(Err::Incomplete(_)) => panic!("Incomplete should never happen"),
+            Ok(y) => Ok(y),
+        }
+    }
+}
+
+/// Similar code than `or_fail()` yet usage and behavior differ:
+/// primarly it is non-blockant as it does not turn Errors into Failures
+/// but rather gives it a PError context (E parameter) therefore updating the generic Nom::ErrorKind
+pub fn or_err<'src, O, F, E>(f: F, e: E) -> impl Fn(PInput<'src>) -> PResult<'src, O>
+where
+    F: Fn(PInput<'src>) -> PResult<'src, O>,
+    E: Fn() -> PErrorKind<PInput<'src>>,
+{
+    move |input| {
+        match f(input) {
+            Err(Err::Failure(err)) => match err.kind {
+                PErrorKind::Nom(_) => Err(Err::Failure(PError {
+                    context: None,
+                    kind: e(),
```

Comment:
i'm not sure we should override the error kind here, could you give an example ?

Reply:
Do you mean that it is just superfuous code since anyway `or_fail()` will be called in case of a failure ?

Reply:
fixed, it was superfluous indeed